src/App.jsx

```diff
@@ -14,45 +15,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FILD_BY_ALPHABET = 'SORT_FILD_BY_ALPHABET';
+const SORT_BY_LENGTH = 'SORT_BY_LENGTH';
+
+function getPrepeadGoods(goods, sortFiled, reverse) {
+  const prepearedGoods = [...goods];
+
+  if (sortFiled) {
+    prepearedGoods.sort((a, b) => {
+      switch (sortFiled) {
+        case SORT_FILD_BY_ALPHABET:
+          return a.nameGood.localeCompare(b.nameGood);
+        case SORT_BY_LENGTH:
+          return a.nameGood.length - b.nameGood.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    return prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const goodsFromServerObjekt = goodsFromServer
+    .map((item, index) => ({ nameGood: item, id: index + 1 }));
+
+  const [sortBtnFild, setSortBtnFild] = useState('');
+  const [reverseList, setReversedList] = useState(false);
+
+  const goodsList = getPrepeadGoods(
+    goodsFromServerObjekt,
+    sortBtnFild,
+    reverseList,
+  );
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const showResetBtn = (sortBtnFild !== '') || reverseList;
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortBtnFild !== SORT_FILD_BY_ALPHABET && 'is-light'} `}
```

Comment:
will be better if you use classnames utils

Reply:
➕